### PR TITLE
Envoy update 1.22.6

### PR DESCRIPTION
**Pull request by jrajahalme**
Update Envoy to 1.22.6.

### CODE REVIEW

File: envoy.bazelrc [hunk 1]
```diff
@@ -42,6 +45,9 @@ build --action_env=CXX
 build --action_env=LLVM_CONFIG
 build --action_env=PATH
 
+# Disable ICU linking for googleurl.
+build --@com_googlesource_googleurl//build_config:system_icu=0
+
```
Comment (sayboras) [inline envoy.bazelrc:50]: What could the the benefit of disabling ICU linking compared to installing libicu-dev package in builder docker?

**Review by sayboras — APPROVED** [review]
This is superb :100:

**Comment by sayboras** [comment]
Cilium CI run is available in https://github.com/cilium/cilium/pull/23502

**Review by tommyp1ckles — APPROVED** [review]
nice!

File: envoy.bazelrc (hunk 1 above)
Comment (jrajahalme) [reply envoy.bazelrc:50]: No idea, I just synced `envoy.bazelrc` to the upstream Envoy `.bazelrc`. We keep our own settings at `.bazelrc` that first includes `envoy.bazelrc`.

File: cilium/grpc_subscription.cc [hunk 2]
```diff
@@ -120,7 +134,8 @@ std::unique_ptr<Config::GrpcSubscriptionImpl> subscribe(
           dispatcher, sotwGrpcMethod(type_url),
           random, scope,
           Config::Utility::parseRateLimitSettings(api_config_source),
-          api_config_source.set_node_on_first_message_only()),
+          api_config_source.set_node_on_first_message_only(),
+	  std::move(nop_config_validators)),
```
[the remark this answers is not in the document]
Comment (jrajahalme) [reply cilium/grpc_subscription.cc:138]: This is one of the internal API changes between 1.21 and 1.22. Not sure if the `std::move()` here is needed, but follows a pattern elsewhere in the Envoy code base for this function.

**Comment by jrajahalme** [comment]
Will have to revert the `pull_request_target:` change from here before merging.

**Comment by jrajahalme** [comment]
Changed to cross-compile with clang-15 on Ubuntu 20.04, so that glibc hacks are not needed.

**Comment by jrajahalme** [comment]
Now builds in ten minutes with the updated build archive :-)

**Comment by jrajahalme** [comment]
Fixed unit test issue surfaced by clang 15.

**Comment by jrajahalme** [comment]
Reverted the `pull_request_target` change to ready this for merging.

**Pull request merged**